### DISCUSSION TITLE
Class based database migrations instead of schema based?

### DISCUSSION THREAD

**Discussion by Drakota** [prisma/prisma#3656] [Q&A] [answered]
Hi everyone,

Has Prisma consired using classes for the database representation (ala TypeORM) instead of the Prisma scheme? In cases like APIs made with `type-graphql` where you use can use the entity class to flag which fields in your database you want to expose in your API, it would be hugely beneficial. 

I really like Primsa and think it is the best experience you could have with an ORM in Node, but I do miss my classes.

Obviously this is not a easy task, but I thought I'd drop a discussion to see if anyone had the same idea.

**Comment by ryands17** [comment, marked as answer]
Hey @Drakota 👋 
What do you think would be the advantage of having a class based database representation? If just exposing the required fields it what you would want as an outcome, you could always use something like [this](https://github.com/MichalLytek/type-graphql/tree/prisma) which auto-generates CRUD operations and also allows customization.

Also, Prisma is not an ORM as we have [here](https://www.prisma.io/docs/understand-prisma/prisma-in-your-stack/is-prisma-an-orm).

If you still feel that there's something missing, do create a feature request here explaining the need for having a class based setup [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=).

Let me know if that helps :)

**Reply by Drakota** [reply]
Thank you for your answer! I was not aware of the branch in question (prisma) and seems to be what I need.

I was also unaware of the approach you guys @ prisma were trying to accomplish with the project and it's difference with a ORM.
An interresting read!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
